### ISSUE TITLE
False positive for unused vars inside decorator arguments 

### ISSUE THREAD

**Issue by yyx990803** [eslint/typescript-eslint-parser#445]
**What version of TypeScript are you using?**
2.6.1

**What version of `typescript-eslint-parser` are you using?**
12.0.0

**What code were you trying to parse?**

```ts
import { Component, Vue } from 'vue-property-decorator';
import HelloWorld from './components/HelloWorld.vue';

@Component({
  components: {
    HelloWorld
  }
})
export default class App extends Vue {}
```

**What did you expect to happen?**
With `eslint-plugin-typescript`'s `typescript/no-unused-vars` rule enabled, should lint without errors.

**What happened?**

```
error: 'HelloWorld' is defined but never used (no-unused-vars)
```

It looks like the expression in a decorator's argument position is not taken into context when determining unused vars. Also, when using `eslint --fix`, code inside decorator arguments are sometimes not formatted correctly.

**Comment by corbinu** [comment]
@yyx990803 I believe this is an issue with https://github.com/nzakas/eslint-plugin-typescript

**Comment by OzoTek** [comment]
Any update on this? I still have this issue, using the most up to date versions of the plugin / parser.

**Comment by JamesHenry** [comment]
Thanks for reporting, @yyx990803! This needs to be addressed in the [plugin](https://github.com/nzakas/eslint-plugin-typescript). It would be fantastic if one of the people affected by this issue could contribute to it.

**Comment by JamesHenry** [comment]
Closing as this is really a bug report for the custom rule in the plugin and an issue has already been opened there. Thanks!
